mlrun/utils/helpers.py

```diff
@@ -1066,6 +1066,27 @@ def is_legacy_artifact(artifact):
         return not hasattr(artifact, "metadata")
 
 
+def get_in_artifact(artifact, key, default=None):
+    if is_legacy_artifact(artifact):
+        return artifact.get(key, default)
+    else:
+        try:
```

Comment:
It would be more elegant to do something like:
```python
if hasattr(artifact.metadata, key):
    return artifact.metadata.key or default
```
Rather than blindly throwing exceptions...